Penteract/Assets/Scripts/PlayerController.cpp

```diff
@@ -187,51 +210,58 @@ void PlayerController::SwitchCharacter() {
 			fang->Enable();
 			hudControllerScript->UpdateHP(lifePointsFang, lifePointsOni);
 		}
-		switchCooldownRemaing = switchCooldown;
+		switchCooldownRemaining = switchCooldown;
 		if (hudControllerScript) {
 			hudControllerScript->ChangePlayerHUD();
 		}
 	}
 }
 
 bool PlayerController::CanShoot() {
-	return !shooting && ((fang && fangCompParticle) || (onimaru && onimaruCompParticle));
+	
+	return !shooting && ((fang->IsActive() && fangTrail) || (onimaru->IsActive() && onimaruTrail));
 }
 
 void PlayerController::Shoot() {
 	ComponentTransform* transform = GetOwner().GetComponent<ComponentTransform>();
-	FangGuntransform = fangGun->GetComponent<ComponentTransform>();
-
+	
 	if (CanShoot()) {
 		if (shootAudioSource) {
```

Comment:
I'd recommend externalizing the Shooting functions in a ProjectileController so you just need to p.eg: ProjectileController:Shoot(direction, speed, etc.) and some logic can be externalized into a specific module